spec/mixlib/install/backend_spec.rb

```diff
@@ -211,6 +293,44 @@
           it_behaves_like "the right artifact info"
         end
       end
+
+      context "when p, pv and m are not present" do
+        context "with a full product version" do
+          let(:product_version) { "12.4.3+20151006083011" }
+          let(:expected_version) { "12.4.3+20151006083011" }
+
+          # This test is currently disabled due to an omnitruck issue.
+          # it_behaves_like "the right artifact list info"
+        end
+
+        context "with a major.minor.patch product version" do
+          let(:product_version) { "12.4.3" }
+          let(:expected_version) { /^12.4.3\+[0-9]{14}$/ }
+
+          it_behaves_like "the right artifact list info"
+        end
+
+        context "with a major.minor product version" do
+          let(:product_version) { "12.4" }
+          let(:expected_version) { /^12.4.\d\+[0-9]{14}$/ }
+
+          it_behaves_like "the right artifact list info"
+        end
+
+        context "with a major product version" do
+          let(:product_version) { "12" }
+          let(:expected_version) { /^12.\d.\d\+[0-9]{14}$/ }
+
+          it_behaves_like "the right artifact list info"
+        end
+
+        context "with latest version keyword" do
+          let(:product_version) { "latest" }
+          let(:expected_version) { /\d.\d.\d/ }
```

Comment:
This regexp should probably look like l:308 `/^12.4.3\+[0-9]{14}$/` (unrelated sorry, but I just noticed it)